### PR TITLE
Map severity, handler, colors

### PR DESCRIPTION
**Pull request by infojunkie**
As per issues #19 and #25.

### CODE REVIEW

File: Slack.php [hunk 1]
```diff
@@ -120,6 +126,20 @@ function bugnote_add_form($event, $bug_id) {
         echo '</td></tr>';
     }
 
+    function get_color($status){
+        // split the status_enum string which is a string like 10:new,20:open, ...
+        $status_array = explode (",", config_get('status_enum_string'));
+        foreach ($status_array as $elem){
+            $elem = explode (":",$elem);
+            // if $status  match the  status number from my config I search the color that match the name of the status 
+            if ($elem[0] == $status){
```
Comment (infojunkie) [inline Slack.php]: needs an additional condition to ensure this entry exists in the `status_colors` settings.

File: Slack.php [hunk 2]
```diff
@@ -340,10 +392,11 @@ function notify($msg, $webhook, $channel, $attachment = FALSE) {
         curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);
         curl_setopt($ch, CURLOPT_TIMEOUT, 5);
         $result = curl_exec($ch);
-        if ($result !== 'ok') {
+        //ignore error to avoid blocking mantis if slack is down
```
Comment (infojunkie) [inline Slack.php]: I'll remove this unrelated change - let's discuss it in a separate issue if needed.

**Comment by infojunkie** [comment]
@AmauryBs here's the pull request and some comments.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,29 @@
-MantisBT-Slack
-==============
+# Changes
+
+MantisBT-Slack forked from https://github.com/infojunkie/MantisBT-Slack
```
Comment (infojunkie) [inline README.md]: Remove the notice about forking.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,29 @@
...
+I added the possibility to map bug severity/bug handler with slack channels.  
+I matched the color displayed in slack with the bug status color.
+
+
+Setup:
+Activate "Use severity" in the plugin configuration pages.
```
Comment (infojunkie) [inline README.md]: Move this documentation to the original "Setup" section below.

File: Slack.php [hunk 5]
```diff
@@ -295,15 +329,33 @@ function format_value($bug, $field_name) {
...
+        
+        if(plugin_config_get("use_handler")){
+            $channels = plugin_config_get('handler_channels');
+            
+            // if there is no channel set for this handler then try to find a channel with severity
+            if (array_key_exists($handler, $channels)){
+            return  $channels[$handler];
+            }
+            
+            if(plugin_config_get("use_handler")){
```
Comment (infojunkie) [inline Slack.php]: shouldn't this be `use_severity` ?

**Comment by TBG-FR** [comment]
Hello ! Did you abort the development of this feature ?

**Comment by infojunkie** [comment]
I closed the PR because the submitter did not follow up on my review. Feel free to reopen and continue the work!

**Pull request closed without merging**